qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+    chemists' order, but rather sorted by index with least index first.
+
+    Args:
+      facs: a list of factors where each element is `(i, c)` where `i` is an integer index and
+        `c` is either `-` or `+`.
+
+    Returns:
+        facs_out: A copy of the reordered factors or the input list (not a copy) if the factors are
+          already in the desired order.
+        phase: Either `1` or `-1`.
+
+    Raises:
+        ValueError: if `facs` does not represent a two-body interaction.
+    """
+    ops = [fac[1] for fac in facs]
+    if ops == ["+", "+", "-", "-"]:
+        facs_out = facs
+        phase = 1
+    elif ops == ["+", "-", "+", "-"]:
+        facs_out = [facs[0], facs[2], facs[1], facs[3]]
+        phase = -1
+    elif ops == ["+", "-", "-", "+"]:
+        facs_out = [facs[0], facs[3], facs[1], facs[2]]
+        phase = 1
+    else:
+        raise ValueError("unexpected sequence of operators", facs)
+    return facs_out, phase
+
+
+class BravyiKitaevSFMapper(FermionicMapper):
+    """The Bravyi-Kitaev super fast fermion-to-qubit mapping.
+
+    This implementation follows closely Reference [1].
+
+       References:
+           [1]: Bravyi-Kitaev Superfast simulation of electronic structure on a quantum computer
+                Kanav Setia and James D Whitfield
+                JCP Volume 148, Issue 16 - Published April 2018
+                arXiv:1712.00446
+    """
+
+    def map(self, second_q_op: FermionicOp) -> PauliSumOp:
+        if not isinstance(second_q_op, FermionicOp):
+            raise TypeError("Type ", type(second_q_op), " not supported.")
+
+        edge_list = bksf_edge_list_fermionic_op(second_q_op)
+        sparse_pauli = _convert_operators(second_q_op, edge_list)
+
+        ## Simplify and sort the result
+        sparse_pauli = sparse_pauli.simplify()
+        indices = sparse_pauli.table.argsort()
+        table = sparse_pauli.table[indices]
+        coeffs = sparse_pauli.coeffs[indices]
+        sorted_sparse_pauli = SparsePauliOp(table, coeffs)
+
+        return PauliSumOp(sorted_sparse_pauli)
+
+
+def _analyze_term(term_str: str) -> Tuple[str, List]:
+    """
+    Return a string recording the type of interaction represented by `term_str` and
+    a list of the factors and their indices in `term_str`.
+
+    The types of interaction are 'number', 'excitation', 'coulomb_exchange', 'number_excitation',
+    'double_excitation'.
+
+    Args:
+       `term_str`: a string of characters in `+-NI`.
+
+    Returns:
+       tuple: The first element is a string specifying the interaction type. See the method
+       `_interaction_type`. The second is a list of factors as returned by `_unpack_term`.
+    """
+    (n_number, n_raise, n_lower), facs = _unpack_term(term_str, expand_number_op=True)
+    ttype = _interaction_type(n_number, n_raise, n_lower)
+    return ttype, facs
+
+
+def _convert_operators(fer_op_qn: FermionicOp, edge_list: np.ndarray) -> SparsePauliOp:
```

Comment:
This method could use some explanatory docstring.

Reply:
done